packages/components/vite.config.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import typescript from '@rollup/plugin-typescript';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { defineConfig } from 'vite';
+
+export default defineConfig(({ mode }) => {
+  return {
+    build: {
+      lib: {
+        entry: path.resolve(__dirname, 'src/index.ts'),
+        // Build ES Module as CommonJS versions.
+        formats: ['es', 'cjs'],
+        fileName: (format, entryName) => {
+          // Since package.json contains `type: "module"`, .js files will be interpreted as ES Modules.
+          // CommonJS modules must be distinguished with a .cjs extension.
+          const extension = { es: 'js', cjs: 'cjs' }[format];
+          // Base output directory on the the module format.
+          // ES Modules are written to lib/es, CommonJS to lib/cjs.
+          return `${format}/${entryName}.${extension}`;
+        },
+      },
+      // Leave minification to consumers.
+      minify: false,
+      // Base output directory. Output paths are based on this and build.fileName.
+      outDir: 'lib',
+      rollupOptions: {
+        // Do not bundle dependencies.
+        external: /node_modules/,
+        output: {
+          paths: (id) => {
+            // ES Modules require imports to be fully specified (including extensions).
+            // Thus, we need to specify the extension for the fp.js file imported from lodash.
+            // This allows us to omit it in source files as we've done in the past in Terra UI.
```

Comment:
Alternatively, we could probably use the [`import/extensions`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md) ESLint and add extensions to relative imports within packages as well.